### PR TITLE
Update PR template to provide guidance for testing

### PR DESCRIPTION
**Pull request by lanej0**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

We would like to provide more guidance for writing testing instructions for pull requests and make the process of opting out of providing testing instructions explicit -- a PR author has to check a box to indicate that they are omitting instructions. This PR tweaks the template used on GitHub when opening a new PR.

### How to test the changes in this Pull Request:

1. Pull this branch
2. Create a new branch based on this branch
3. Make a minor change and push it
4. Open a pull request
5. Note the changes to the template

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by lanej0** [comment]
Apparently, changes to the PR template can't be tested as I've outlined in the testing instructions.  Best to just double-check the Markdown.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -14,17 +14,25 @@
...
 3.
 
+<!-- End testing instructions -->
+
 ### Other information:
 
 -   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
 -   [ ] Have you written new tests for your changes, as applicable?
--   [ ] Have you successfully run tests with your changes locally?
+-   [ ] Have you successfully run tests with your changes locally (including end to end tests)?
```
Comment (tammullen) [inline .github/PULL_REQUEST_TEMPLATE.md]: As the tests are now required in CI, I wonder is this task needed, or perhaps we can reword it to focus on testing of the changes themselves.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -14,17 +14,25 @@
 
 Closes # .
 
+<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions.  Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test.-->
```
Comment (tammullen) [inline .github/PULL_REQUEST_TEMPLATE.md]: To match the location of the other comments in the template, maybe this should be `This section is mandatory` and the comments and new task moved to the `### How to test the changes in this Pull Request:` section

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (lanej0) [reply .github/PULL_REQUEST_TEMPLATE.md]: There's occasionally differences in results between CI and locally, but hopefully that will be less and less as we improve the test suite.  So I agree with you on this point, let's remove it.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (lanej0) [reply .github/PULL_REQUEST_TEMPLATE.md]: Good call.

**Comment by alopezari** [comment]
> Apparently, changes to the PR template can't be tested as I've outlined in the testing instructions. Best to just double-check the Markdown.

Hi @lanej0! Just wanted to confirm that you can test this PR. I did the following:
1. Fork WooCommerce repository.
2. Cd to your forked repo and run `git fetch upstream` to bring this branch.
3. Checkout to this branch in the forked repo and merge it into `trunk`.
4. Create a new branch, add any change and create a PR.
5. Check the template looks as expected:
Screenshot 2022-09-21 at 11 30 34

Let me know if the screenshot above looks as expected! In that case, I'll finish the review and approve the PR.

**Comment by lanej0** [comment]
@woocommerce/atlas Pinging you for awareness/to see if you have any concerns?

**Comment by rodelgc** [comment]
There seems to be a recent merge conflict in `PULL_REQUEST_TEMPLATE.md`:

**Comment by rodelgc** [comment]
Looking good. The added sections appear correctly in both the edit mode and preview mode on my fork:

On edit mode:

Screen Shot 2022-10-18 at 8 41 17 PM

Preview:
Screenshot 2022-10-18 at 20-42-38 Comparing trunk try_new-pr-template · rodelgc_woocommerce

**Review by samueljseay — DISMISSED** [review]
This looks good to me. A nice addition to make sure people think harder about testing PRs.

**Review by rodelgc — APPROVED** [review]
